### PR TITLE
Fix for issue with Default material not available when Models are imported at project startup.

### PR DESCRIPTION
**Pull request by Thomasch-unity3d**
### **Please read**
**PR workflow guidelines**
* SRP ABV will start automatically on Yamato when you open your PR
* Changes to docs and md files will **not** trigger ABV jobs 
* Consider making use of **draft PRs** if you are not 100% sure that your PR is ready for review
* ABV will restart if you add a new commit to a branch with an open PR (hence why you should consider using draft PRs)
* Adding [skip ci] (case insensitive) to the title of PRs will stop any jobs being trigger automatically - you will need to open Yamato and find your branch to run ABV
* You can also add [skip ci] to commit messages to prevent CI from running on that push
* Add [cancel old ci] to your commit message if you've made changes you want to test and no longer need the previous jobs

### Checklist for PR maker
- [ ] Have you added a backport label (if needed)? For example, the `need-backport-*` label. After you backport the PR, the label changes to `backported-*`.
- [ ] Have you updated the changelog? Each package has a `CHANGELOG.md` file.
- [ ] Have you updated or added the documentation for your PR? When you add a new feature, change a property name, or change the behavior of a feature, it's best practice to include related documentation changes in the same PR.
- [ ] Have you added a graphic test for your PR (if needed)? When you add a new feature, or discover a bug that tests don't cover, please add a graphic test.

---
### Purpose of this PR

Fixes case 1141488 where Materials generated by the model importer fallback to the legacy default material rather than HDRP's default material.
The issue here is that HDRenderPipelineEditorResources fails to load using LoadAssetAtPath.
Using LoadSerializedFileAndForget succeeds but imples that changes to the HDRenderPipelineEditorResources asset are not picked up until Unity is restarted.

---
### Testing status

**Manual Tests**: What did you do?
- [ ] Opened test project + Run graphic tests locally
- [ ] Built a player
- [ ] Checked new UI names with UX convention
- [ ] Tested UI multi-edition + Undo/Redo + Prefab overrides + Alignment in Preset
- [ ] C# and shader warnings (supress shader cache to see them)
- [ ] Checked new resources path for the reloader (in developer mode, you have a button at end of resources that check the paths)
- Other: 

**Automated Tests**: What did you setup? (Add a screenshot or the reference image of the test please)

**Yamato**: (Select your branch):
https://yamato.prd.cds.internal.unity3d.com/jobs/902-Graphics

Any test projects to go with this to help reviewers?

---
### Comments to reviewers
Notes for the reviewers you have assigned.

### CODE REVIEW

File: com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs [hunk 1]
```diff
@@ -542,9 +546,19 @@ void UpgradeResourcesIfNeeded()
...
+                var objs = InternalEditorUtility.LoadSerializedFileAndForget(HDUtils.GetHDRenderPipelinePath() + "Editor/RenderPipelineResources/HDRenderPipelineEditorResources.asset");
+                HDRenderPipeline.defaultAsset.renderPipelineEditorResources = objs != null && objs.Length > 0 ? objs.First() as HDRenderPipelineEditorResources : null;
+            }
+
+            if (ResourceReloader.ReloadAllNullIn(HDRenderPipeline.defaultAsset.renderPipelineEditorResources,
+                HDUtils.GetHDRenderPipelinePath()))
+            {
+                InternalEditorUtility.SaveToSerializedFileAndForget(
+                    new Object[]{HDRenderPipeline.defaultAsset.renderPipelineEditorResources },
+                    HDUtils.GetHDRenderPipelinePath() + "Editor/RenderPipelineResources/HDRenderPipelineEditorResources.asset",
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Runtime/RenderPipeline/HDRenderPipeline.cs:559]: Same

File: com.unity.render-pipelines.high-definition/Editor/DefaultScene/HDWizard.Configuration.cs [hunk 2]
```diff
@@ -378,9 +379,16 @@ void FixHdrpAssetRuntimeResources(bool fromAsyncUnused)
...
-                = AssetDatabase.LoadAssetAtPath<RenderPipelineResources>(HDUtils.GetHDRenderPipelinePath() + "Runtime/RenderPipelineResources/HDRenderPipelineResources.asset");
-            ResourceReloader.ReloadAllNullIn(HDRenderPipeline.defaultAsset.renderPipelineResources, HDUtils.GetHDRenderPipelinePath());
+            var objs = InternalEditorUtility.LoadSerializedFileAndForget(HDUtils.GetHDRenderPipelinePath() + "Editor/RenderPipelineResources/HDRenderPipelineEditorResources.asset");
+            hdrpAsset.renderPipelineEditorResources = objs != null && objs.Length > 0 ? objs.First() as HDRenderPipelineEditorResources : null;
+            if (ResourceReloader.ReloadAllNullIn(HDRenderPipeline.defaultAsset.renderPipelineEditorResources,
+                HDUtils.GetHDRenderPipelinePath()))
+            {
+                InternalEditorUtility.SaveToSerializedFileAndForget(
+                    new UnityEngine.Object[] { HDRenderPipeline.defaultAsset.renderPipelineEditorResources },
+                    HDUtils.GetHDRenderPipelinePath() + "Editor/RenderPipelineResources/HDRenderPipelineEditorResources.asset",
```
Comment (RSlysz) [inline com.unity.render-pipelines.high-definition/Editor/DefaultScene/HDWizard.Configuration.cs:389]: Can you set a const for the several time used path?

**Comment by sebastienlagarde** [comment]
Thanks for this @Thomasch-unity3d . I have merged master and relaunched as all was red but not due to this PR. Question what are we donig with this PR: https://github.com/Unity-Technologies/ScriptableRenderPipeline/pull/5862 (Guess it is almost the same?) could you close it. THanks!

**Comment by Thomasch-unity3d** [comment]
Yes,  Unity-Technologies/ScriptableRenderPipeline#5862 is the previous iteration, closed now. QA is in progress but I'd be great to have more from your team and there might be other issues that this fixes ( ie : I know there's a similar bug for speedTree in URP ). Thanks guys!

**Comment by Thomasch-unity3d** [comment]
And just to make sure, are you ok with " changes to the HDRenderPipelineEditorResources asset are not picked up until Unity is restarted." ? It's supposed to be read-only for users so I assume it's not a problem but should we document that ?

**Comment by sebastienlagarde** [comment]
I am adding our QA so they can check. @Thomasch-unity3d could you put a description about what you have tested and what workflow was failing before this PR. THanks (for @Unity-Technologies/gfx-qa-hdrp  Thomas ios in Montreal timezone)

**Comment by sebastienlagarde** [comment]
One of the thing to pay attention is upgrade (between two version where we have added stuff in RPResources, this could be from 8.1.0 to latest 9.x-preview package (so this PR))

**Comment by iM0ve** [comment]
I also upgraded 2 project from HDRP 8.0.0 Unity 2020.1.a23 to this PR Unity 2020.2.a11

Visually there were no issues and no pink materials, however noticed a new minor issue. The yellow highlight stopped working that showcases which HDRP editor resources asset is used. See the video

HDRP/staging (working)
Highlights and selects the asset

staging

hdrp-editorResourcesAsset-loadAndForget (issue)
Does not highlight only selects

PR

**Comment by Thomasch-unity3d** [comment]
Thanks for the test pass, I'm not sure how to deal with that issue, that field is editable but any change in there is reverted when the editor is closed and restarted because HDRenderPipelineAsset always finds the HDRenderPipelineEditorResources.asset at a specific hardcoded path.
I wonder if that field should be removed or made not editable, I'm confused about its role. I'll see if I can force the ObjectField to always highlight that asset.
Any other idea or suggestion ?

**Comment by sebastienlagarde** [comment]
> Thanks for the test pass, I'm not sure how to deal with that issue, that field is editable but any change in there is reverted when the editor is closed and restarted because HDRenderPipelineAsset always finds the HDRenderPipelineEditorResources.asset at a specific hardcoded path.
> I wonder if that field should be removed or made not editable, I'm confused about its role. I'll see if I can force the ObjectField to always highlight that asset.
> Any other idea or suggestion ?

@RSlysz any thought? thanks

**Comment by RSlysz** [comment]
True... I haven't see this comming.
Without a loading persistent (with an handle) you cannot ping it any more in the Project Window.

In therm of memory management, the current Resource Asset is saved on the only occasion where it can be modified so it is good.

In therm of loading issue, it seams to fix somewhat issue that was happening previously.

So I can only see one way. As using this loading method was a workarround in the first place, I think that we should aim to reload with persistency some frame later. This way, the ping will be working again. 
@Thomasch-unity3d : Do you see any time after the initial loading where you can perform a standard load again ?

**Pull request merged**